src/parser/program.rs

```diff
@@ -133,26 +136,40 @@ impl ParsedConfig {
     }
 }
 
-impl From<ParsedConfig> for Config {
-    fn from(origin: ParsedConfig) -> Self {
-        Self {
-            programs: origin
-                .programs
-                .into_iter()
-                .fold(vec![], |mut acc, (key, value)| {
-                    acc.push(Program::from(key, value));
-                    acc
-                }),
+impl TryFrom<ParsedConfig> for Config {
+    type Error = ParseError;
+
+    fn try_from(origin: ParsedConfig) -> Result<Self, ParseError> {
+        let mut programs = Vec::new();
+        for (_, value) in origin.programs {
+            programs.push(Program::try_from(value)?);
         }
+        Ok(Config { programs })
     }
 }
 
-impl Program {
-    fn from(name: String, origin: ParsedProgram) -> Self {
-        Self {
-            name,
+impl TryFrom<ParsedProgram> for Program {
+    type Error = ParseError;
+
+    fn try_from(origin: ParsedProgram) -> Result<Self, ParseError> {
+        let umask_str = origin.umask.clone().unwrap_or_else(|| String::from("000"));
+        let umask = u32::from_str_radix(&umask_str, 8).map_err(|_| {
+            ParseError::InvalidUmask(
+                "Invalid umask".to_string(),
+                origin.name.clone().unwrap_or_else(|| String::from("")),
+            )
+        })?;
+        if umask > 512 {
```

Comment:
Shouldn't this be `umask > 511` or `umask >= 512`?
Also could write it in octal instead (`umask > 0o777`)
And it would be nice for the validation of the umask to be in a function